ThunkLibs/libxcb/libxcb_Guest.cpp

```diff
@@ -22,71 +22,35 @@ tags: thunklibs|xcb
 
 #include "common/Guest.h"
 #include "WorkEventData.h"
-#include "common/CrossArchEvent.h"
 
 #include <stdarg.h>
 
 #include "thunkgen_guest_libxcb.inl"
 
-static std::thread CBThread{};
-static std::atomic<bool> CBDone{false};
+struct OnStart {
+std::thread thr;
+std::atomic<bool> done { false };
+std::mutex m;
+std::condition_variable var;
 
-static CrossArchEvent WaitForWork{};
-static CrossArchEvent WorkDone{};
-static CBWork CBWorkData{};
+OnStart() : thr([this]() {
+struct { unsigned id = 1; } args;
+fexthunks_fex_register_async_worker_thread(&args);
 
-void FEX_Helper_GiveEvents() {
-  struct {
-    CrossArchEvent *WaitForWork;
-    CrossArchEvent *WorkDone;
-    CBWork *Work;
-  } args;
+std::unique_lock lock(m);
+var.wait(lock, [this]() -> bool { return done; });
```

Comment:
How does this handle the case that spurious wakeup on mutex will corrupt the stack of the running thread when we have stolen its state?

Reply:
Fair point. Maybe this should be moved into a thunk implemented in Thunks.cpp, which would do the waiting purely outside the VM and which wouldn't return until shutdown?

Reply:
That will...probably work

Reply:
Done, still passes my artificial test program.